torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp

```diff
@@ -266,12 +261,23 @@ void ThreadPredicateMap::updateBitSet(const Expr* expr) {
       }
     }
 
+    // Figure out which dims bcast wants to reset
+    auto this_input_preds = pred_and_src.limited_types;
```

Comment:
This change makes sense to me and I'm not against it. I agree that we probably never want any input to reset other inputs' predicate, hard to imagine : ). 

But just out of curiosity, would the original method be good enough if we just `go through all the inputs, and make sure no input is parallelized on an parallel type that another one predicates on` before any predicate propagation?

Reply:
`no input is parallelized on an parallel type that another one predicates on`: <- this can be actually valid if shared memory is used and the corresponding output domain is not parallelized. 

I'm not sure the original method could be made work correctly. To me, the way in this PR seems to make most sense.

Reply:
I see, makes sense. The shared mem cases could be tricky on the original one. Thanks for the clarification.